src/definitions/modules/modal.js

```diff
@@ -806,18 +803,25 @@ $.fn.modal = function(parameters) {
             }
           },
           modalOffset: function() {
-            var
-              width = module.cache.width,
-              height = module.cache.height
-            ;
-            $module
-              .css({
-                marginTop: (!$module.hasClass('aligned') && module.can.fit())
-                  ? -(height / 2)
-                  : 0,
-                marginLeft: -(width / 2)
-              })
-            ;
+            if (!settings.detachable) {
+              $module
+                .css({
+                  top: (!$module.hasClass('aligned') && module.can.fit())
+                    ? parseInt(($(window).height() / 2) + $(document).scrollTop() - ($module.height() - settings.padding))
+                    : $(document).scrollTop() + (settings.padding / 2),
+                  marginLeft: -(module.cache.width / 2)
+                }) 
+              ;
```

Comment:
Unfortunately this part is somehow not correctly working when the modal is bigger and the background body has less content
See http://jsfiddle.net/bkh17Lgy/
Without this PR change that example was working http://jsfiddle.net/bkh17Lgy/1/

@prudho Could you please take a look again at this and prepare another PR? 😉

Reply:
Opened #1079 for this